lib/mayaUsd/utils/utilFileSystem.cpp

```diff
@@ -113,3 +115,64 @@ std::string UsdMayaUtilFileSystem::resolveRelativePathWithinMayaContext(
 
     return path.string();
 }
+
+bool UsdMayaUtilFileSystem::isDirectory(const char* dirPath)
+{
+    // TODO: (yliangsiew) Is there a platform wrapper somewhere so I can implement the OS native
+    // calls for something simple like this without having to add yet another dependency on Boost?
```

Comment:
Yes, I noticed that having std::filesystem available would simplify this pull request a lot. I agree that boost::filesystem is not an option since it has diverged with the C++17 standard in very noticeable ways. Currently looking for a way to get rid of this compilation unit before it becomes too large.